### PR TITLE
Enhancement: useChartSelection composition

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Similar to `useChartCursor` adds a `useChartSelection` composition for sharing selection state

### CODE REVIEW

**Review by znicholasbrown — APPROVED** [review]
(non-blocking) in the context of a `chartSelection`, I don't think `selection` is adding any value, I'd change the payload fields to `start` and `end` for brevity

File: src/compositions/useChartSelection/useChartSelection.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { ref, Ref } from 'vue'
+
+export type UseChartSelection = {
+  selectionStart: Ref<Date | null>,
+  selectionEnd: Ref<Date | null>,
+}
+
+function selectionFactory(): UseChartSelection {
+  const selectionStart = ref<Date | null>(null)
+  const selectionEnd = ref<Date | null>(null)
```
Comment (znicholasbrown) [inline src/compositions/useChartSelection/useChartSelection.ts:10]: I'm guessing it's ok that start and end don't have any enforcement of linearity?

Comment (pleek91) [reply src/compositions/useChartSelection/useChartSelection.ts:10]: Correct, the ChartSelection component handles that. You can technically select backwards from right to left or forwards from left to right.

Comment (znicholasbrown) [reply src/compositions/useChartSelection/useChartSelection.ts:10]: Ok I buy that - thanks for the explanation

**Comment by pleek91** [comment]
> 

I do agree with this. But I want to keep these consistent with the component which uses `selectionStart` and `selectionEnd`. Maybe at some point I'll remove the selection prefix from both.

**Pull request merged**